table/block_based/block_based_table_factory.cc

```diff
@@ -396,36 +378,61 @@ static std::unordered_map<std::string, OptionTypeInfo>
             return Cache::CreateFromString(opts, value, cache);
           }}},
         {"block_cache_compressed",
-         {0, OptionType::kUnknown, OptionVerificationType::kDeprecated,
-          OptionTypeFlags::kNone}},
+         {0, OptionType::kUnknown, OptionVerificationType::kDeprecated}},
         {"max_auto_readahead_size",
          {offsetof(struct BlockBasedTableOptions, max_auto_readahead_size),
-          OptionType::kSizeT, OptionVerificationType::kNormal,
-          OptionTypeFlags::kMutable}},
+          OptionType::kSizeT, OptionVerificationType::kNormal}},
         {"prepopulate_block_cache",
          OptionTypeInfo::Enum<BlockBasedTableOptions::PrepopulateBlockCache>(
              offsetof(struct BlockBasedTableOptions, prepopulate_block_cache),
-             &block_base_table_prepopulate_block_cache_string_map,
-             OptionTypeFlags::kMutable)},
+             &block_base_table_prepopulate_block_cache_string_map)},
         {"initial_auto_readahead_size",
          {offsetof(struct BlockBasedTableOptions, initial_auto_readahead_size),
-          OptionType::kSizeT, OptionVerificationType::kNormal,
-          OptionTypeFlags::kMutable}},
+          OptionType::kSizeT, OptionVerificationType::kNormal}},
         {"num_file_reads_for_auto_readahead",
          {offsetof(struct BlockBasedTableOptions,
                    num_file_reads_for_auto_readahead),
-          OptionType::kUInt64T, OptionVerificationType::kNormal,
-          OptionTypeFlags::kMutable}},
+          OptionType::kUInt64T, OptionVerificationType::kNormal}},
+    };
+
+    for (auto& i : info) {
+      switch (i.second.GetType()) {
+        case OptionType::kString:
+        case OptionType::kEnv:
+        case OptionType::kStruct:
+        case OptionType::kVector:
+        case OptionType::kConfigurable:
+        case OptionType::kCustomizable:
+        case OptionType::kEncodedString:
+        case OptionType::kArray:
+        case OptionType::kUnknown:
+          // Not mutable, at least until race condition is resolved
+          break;
+        default:
+          if (i.first == "no_block_cache") {
+            // Special case of not being mutable; only makes sense at
+            // initialization time, and for non-expert users
+          } else {
+            // Values up to 64 bits should be generally safe to mutate despite
+            // the read-write race.
+            i.second.SetFlags(i.second.GetFlags() | OptionTypeFlags::kMutable);
+          }
+          break;
+      }
+    }
 
-};
+    // Exceptions to mutability:
```

Comment:
Is this line needed in case the default OptionTypeFlags can change to not kNone?

Reply:
Oops, left over from previous code revision